lib/gadget/gadget.vala

```diff
@@ -494,6 +494,25 @@ namespace Frida.Gadget {
 	private Config load_config (Location location) throws Error {
 		var config_path = derive_config_path_from_file_path (location.path);
 
+		if(FileUtils.test (config_path, FileTest.EXISTS)) {
```

Comment:
Also this should go in `#if ANDROID` as checking whether the file exists isn't useful on the other platforms.